src/main/java/net/sf/jabref/exporter/FileSaveSession.java

```diff
@@ -0,0 +1,144 @@
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.io.FileBasedLock;
+import net.sf.jabref.logic.util.io.FileUtil;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Class used to handle safe storage to disk.
+ * <p>
+ * Usage: create a SaveSession giving the file to save to, the encoding, and whether to make a backup. The SaveSession
+ * will provide a Writer to store to, which actually goes to a temporary file. The Writer keeps track of whether all
+ * characters could be saved, and if not, which characters were not encodable.
+ * <p>
+ * After saving is finished, the client should close the Writer. If the save should be put into effect, call commit(),
+ * otherwise call cancel(). When canceling, the temporary file is simply deleted and the target file remains unchanged.
+ * When committing, the temporary file is copied to the target file after making a backup if requested and if the target
+ * file already existed, and finally the temporary file is deleted.
+ * <p>
+ * If committing fails, the temporary file will not be deleted.
+ */
+public class FileSaveSession extends SaveSession {
+
+    private static final Log LOGGER = LogFactory.getLog(FileSaveSession.class);
+
+    // Filenames.
+    private static final String BACKUP_EXTENSION = ".bak";
+    private static final String TEMP_PREFIX = "jabref";
+    private static final String TEMP_SUFFIX = "save.bib";
+    private final boolean useLockFile;
+    private Path temporaryFile;
+
+    public FileSaveSession(Charset encoding, boolean backup) throws SaveException {
+        this(encoding, backup, createTemporaryFile());
+    }
+
+    public FileSaveSession(Charset encoding, boolean backup, Path temporaryFile) throws SaveException {
+        super(encoding, backup, getWriterForFile(encoding, temporaryFile));
+        this.temporaryFile = temporaryFile;
+        this.useLockFile = Globals.prefs.getBoolean(JabRefPreferences.USE_LOCK_FILES);
+    }
+
+    private static VerifyingWriter getWriterForFile(Charset encoding, Path file) throws SaveException {
+        try {
+            return new VerifyingWriter(new FileOutputStream(file.toFile()), encoding);
+        } catch (FileNotFoundException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    private static Path createTemporaryFile() throws SaveException {
+        try {
+            return Files.createTempFile(FileSaveSession.TEMP_PREFIX, FileSaveSession.TEMP_SUFFIX);
+        } catch (IOException e) {
+            throw new SaveException(e);
+        }
+    }
+
+    @Override
+    public void commit(File file) throws SaveException {
```

Comment:
Use nio methods 
e.g. File.exist
and  sth like Path.get(path, name, backup Extension)